src/Commands/core/LoginCommands.php

```diff
@@ -24,40 +29,108 @@ class LoginCommands extends DrushCommands
      *   Open firefox web browser, login as the user with the e-mail address drush@example.org and redirect to the path admin/settings/performance.
      */
-    public function login($path = '', $options = ['name' => '1', 'browser' => '', 'redirect-port' => ''])
-    {
-
-        // Redispatch if called against a remote-host so a browser is started on the
-        // the *local* machine.
-        $alias = drush_get_context('DRUSH_TARGET_SITE_ALIAS');
-        if (drush_sitealias_is_remote_site($alias)) {
-            $return = drush_invoke_process($alias, 'user-login', $options['name'], drush_redispatch_get_options(), array('integrate' => false));
-            if ($return['error_status']) {
-                throw new \Exception('Unable to execute user login.');
-            } else {
-                $link = is_string($return['object']) ?: current($return['object']);
-            }
-        } else {
-            if (!drush_bootstrap(DRUSH_BOOTSTRAP_DRUPAL_FULL)) {
-                // Fail gracefully if unable to bootstrap Drupal.
-                // drush_bootstrap() has already logged an error.
-                return false;
-            }
+    public function login($user = '', $path = '', $options = ['uid' => '', 'name' => '', 'mail' => '', 'browser' => '', 'redirect-port' => '']) {
 
-            if ($options['name'] == 1) {
-                $account = User::load(1);
-            } elseif (!$account = user_load_by_name($options['name'])) {
-                throw new \Exception(dt('Unable to load user: !user', array('!user' => $options['name'])));
-            }
-            $link = user_pass_reset_url($account). '/login';
-            if ($path) {
-                $link .= '?destination=' . $path;
+      // Redispatch if called against a remote-host so a browser is started on the
+      // the *local* machine.
+      $alias = drush_get_context('DRUSH_TARGET_SITE_ALIAS');
+      if (drush_sitealias_is_remote_site($alias)) {
+        $return = drush_invoke_process($alias, 'user-login', [$user, $path], drush_redispatch_get_options(), ['integrate' => FALSE]);
+        if ($return['error_status']) {
+          throw new \Exception('Unable to execute user login.');
+        }
+        else {
+          $link = is_string($return['object']) ?: current($return['object']);
+        }
+      }
+      else {
+        if (!drush_bootstrap(DRUSH_BOOTSTRAP_DRUPAL_FULL)) {
+          // Fail gracefully if unable to bootstrap Drupal.
+          // drush_bootstrap() has already logged an error.
+          return FALSE;
+        }
+
+        // Convenience variables.
+        $account = $uid = $name = $mail = NULL;
+        $user_as_path = FALSE;
+
+        // Has the user attempted to specify a target user through options?
+        $has_user_option = (!empty($options['uid']) || !empty($options['name']) || !empty($options['mail']));
+
+        if ($has_user_option) {
```

Comment:
`drush uli /path` should be interpreted as a path, not a user, even if there is no user option.